src/Collections-Strings/String.class.st

```diff
@@ -2649,26 +2649,25 @@ String >> withNoLineLongerThan: aNumber [
 
 { #category : #converting }
 String >> withSeparatorsCompacted [
-	"Return a copy of the receiver with each sequence of whitespace (separator) 
-	characters replaced by a single space character"
-
-	self isEmpty ifTrue: [ ^ self ].
-	^ self class new: self size streamContents: [:stream | 
-		| lastBlank |
-		lastBlank := false.
-		self do: [ :eachChar | 
-			lastBlank 
-				ifTrue: [
-					(lastBlank := eachChar isSeparator)
-						ifFalse: [ stream nextPut: eachChar ] ] 
-				ifFalse: [
-					(lastBlank := eachChar isSeparator)
-						ifTrue: [ stream nextPut: $  ]
-						ifFalse: [ stream nextPut: eachChar ] ] ] ]
-
-	"' test ' withSeparatorsCompacted"
-	"' test test' withSeparatorsCompacted"
-	"'test  test		 ' withSeparatorsCompacted"
+    "Returns a copy of the receiver with each sequence of whitespace (separator) 
+    characters replaced by a single space character"
+    "' test ' withSeparatorsCompacted >>> ' test '"
+    "' test  test' withSeparatorsCompacted >>> ' test test'"
+    "'test  test      ' withSeparatorsCompacted >>> 'test test '"
+
+    self isEmpty ifTrue: [ ^ self ].
```

Comment:
```suggestion
    self ifEmpty: [ ^ self ].
```

Reply:
Often in the kernel classes, those kind of shortcuts are omitted. I assume because of some inlining. I'll wait for Marcus to approve this as well =)

Reply:
ifEmpty is a bit odd... some people feel that it goes to far and do not use it...  we I think never added a rule to enforce using it due to that.

Reply:
for inlining: isEmpty IifTrue: is a message send + inlined ifTrue:, so it indeed is faster.